### ISSUE TITLE
Better Explanation for Security Middleware Section

### ISSUE THREAD

**Issue by Leyart** [krakenjs/swaggerize-express#67]
I cannot make the "Security Middleware" work for my specification. 
My security definition is the following in YAML:

securityDefinitions:
 myauth:
    type: oauth2
    authorizationUrl: "http://api.mywebsite.com/oauth/dialog"
    flow: implicit
    **x-authorize: "authorize"**
    scopes:
      "write:user": "modifies users' information"
      "read:user": "reads users' information"
      "write:request": "needed to write requests on the server"
      "read:request": "authorization to read requests on the server"
      "read:driver": "reads drivers' information"
      "write:driver": "modifies drivers' information"

But I don't think I have added properly the x-authorize parameter.

What does "an appropriate authorize function exists (defined using x-authorize in the securityDefinitions as per swaggerize-routes)" mean? Could it be explained in a detailed way, possibly with an example? 
Thank you in advance,

Luca

**Comment by tlivings** [comment]
Noted. Will add more detail.

Basically, `x-authorize` should point at a path to a file that exports a single middleware (in the case of express) function.

**Comment by Lucg** [comment]
The latest version of Generator Swaggerize is not using the proper swaggerize-express version. Part of my issues were due to this.

**Comment by tlivings** [comment]
Yes - I mis-published something! Corrections coming.

**Comment by talvor** [comment]
The path to the security function, is this relative to the project root, or to the API path?

**Comment by subeeshcbabu** [comment]
The path is relating the project root.
https://github.com/krakenjs/swaggerize-routes/blob/master/lib/buildroutes.js#L161

**Comment by djMax** [comment]
What's the theory behind sticking project specific auth information into a swagger doc? I wouldn't want to expose that path in the external swagger doc, and I don't really want two separate docs for a single service right?

**Comment by tlivings** [comment]
Yeah, probably makes sense to filter out `x-` vendor extensions from the exposes API definition.

**Comment by djMax** [comment]
That's one option, but I'd vastly prefer being able to pass in named authorization handlers to the swaggerize function, because I don't like the idea of embedding "function calls" in what feels like an external facing API document... I'm not saying my approach is the only one, but it seems "reasonable." :)

**Comment by djMax** [comment]
In the meantime, essentially I just load the JSON swagger doc myself and inject the authorization function into the securityDefinitions before passing to swaggerize.

**Comment by tlivings** [comment]
Yeah might make sense to pass `securityDefinitions` as an explicit option so it isn't even in the document. Or pass named auth handlers.
